pkg/daemon/controller/controller.go

```diff
@@ -247,6 +248,11 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 		klog.Error(err)
 		return ctrl.Result{}, err
 	}
+	// Version annotation is the indicator that the node was fully configured by this version of the services ConfigMap
+	err = metadata.ApplyAnnotations(sc.client, sc.ctx, node, map[string]string{metadata.VersionAnnotation: desiredVersion})
```

Comment:
I would rather make `ApplyAnnotations` an internal helper and expose `metadata.ApplyDesiredVersion()` and so on.